tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/nvptx_backend_lib.h

```diff
@@ -29,15 +30,16 @@ limitations under the License.
 namespace xla {
 namespace gpu {
 
+using GpuVersion = absl::variant<std::pair<int, int>, int>;
+
 // Compiles the argument module and returns it. libdevice_dir_path is the parent
```

Comment:
I intend to retain the interface of `CompileToPtx` and keep its comments as-is.

After this PR gets reviewed / revised and merged, I'll submit another PR for AMDGPU-specific logic. On AMDGPU side, a new interface `CompileToHsaco` would be introduced, and it would return `StatusOr<std::vector<uint8>>` as LLVM AMDGPU produces AMD GCN ISA binary in "HSA Code Object (HSACO)" format.